### PR TITLE
📚 Doc: Updates to API documentation and README

### PR DESCRIPTION
**Pull request by gaby**
# Description

- Auto update the API documentation using `o1-mini` from OpenAI
- Auto update the `README` USING `o1-mini` from OpenAI

Related to #2712 

## Changes introduced
- [x] Documentation Update: Detail the updates made to the documentation and links to the changed files.
- [x] Examples: Provide examples demonstrating the new features or changes in action.

## Type of change
- [x] Documentation update (changes to documentation)

### CODE REVIEW

**Comment by gaby** [comment]
@coderabbitai ignore

**Comment by gaby** [comment]
Will fix lint issues in the morning

**Comment by gaby** [comment]
Ready for review :-)

**Review by sixcolors — APPROVED** [review]
LGTM

File: docs/api/app.md [hunk 1]
```diff
@@ -104,90 +135,106 @@ func (app *App) Route(path string) Register
...
+)
 
 func main() {
     app := fiber.New()
 
     app.Get("/john/:age", handler)
     app.Post("/register", handler)
 
-    data, _ := json.MarshalIndent(app.Stack(), "", "  ")
+    data, _err_ := json.MarshalIndent(app.Stack(), "", "  ")
```
Comment (ReneWerner87) [inline docs/api/app.md:234]: not sure about this `_err_`

File: docs/api/app.md [hunk 2]
```diff
@@ -104,90 +135,106 @@ func (app *App) Route(path string) Register
...
+    "encoding/json"
+    "log"
+
+    "github.com/gofiber/fiber/v3"
+)
 
 func main() {
     app := fiber.New()
 
     app.Get("/john/:age", handler)
```
Comment (ReneWerner87) [inline docs/api/app.md:231]: the handler declaration is missing

File: docs/api/app.md [hunk 3]
```diff
@@ -410,173 +488,212 @@ func main() {
...
-Config returns the [app config](./fiber.md#config) as value \( read-only \).
+`Config` returns the [app config](./fiber.md#config) as a value (read-only).
 
 ```go title="Signature"
 func (app *App) Config() Config
 ```
 
 ## Handler
 
-Handler returns the server handler that can be used to serve custom [`\*fasthttp.RequestCtx`](https://pkg.go.dev/github.com/valyala/fasthttp#RequestCtx) requests.
```
Comment (ReneWerner87) [inline docs/api/app.md:421]: the wildcard should be escaped since it is a markdown character

File: docs/api/app.md [hunk 4]
```diff
@@ -410,173 +488,212 @@ func main() {
...
+    Name string `yaml:"name"`
+}
+
+type customBinder struct{}
 
-// My custom binder
-customBinder := &customBinder{}
-// Name of custom binder, which will be used as Bind().Custom("name")
-func (*customBinder) Name() string {
+func (cb *customBinder) Name() string {
```
Comment (ReneWerner87) [inline docs/api/app.md:581]: normally no need for the "cb" string since it is not used by the inner method

File: .github/README.md [hunk 5]
```diff
@@ -606,35 +697,35 @@ Here is a list of middleware that are included within the Fiber framework.
...
-| :------------------------------------------------ | :-------------------------------------------------------------------------------------------------------------------- |
-| [contrib](https://github.com/gofiber/contrib)     | Third party middlewares                                                                                               |
-| [storage](https://github.com/gofiber/storage)     | Premade storage drivers that implement the Storage interface, designed to be used with various Fiber middlewares.     |
-| [template](https://github.com/gofiber/template)   | This package contains 9 template engines that can be used with Fiber `v3` Go version 1.22 or higher is required.      |
+| Middleware                                      | Description                                                                                                           |
+|-------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
```
Comment (ReneWerner87) [inline .github/README.md:708]: hmm not sure
there should be a spce between pipe and hyphen

File: docs/api/bind.md [hunk 6]
```diff
@@ -145,25 +144,25 @@ app.Post("/", func(c fiber.Ctx) error {
...
 
-It is important to specify the correct struct tag based on the content type to be parsed. For example, if you want to parse a MultipartForm body with a field called Pass, you would use a struct field of `form:"pass"`.
+It is important to specify the correct struct tag based on the content type to be parsed. For example, if you want to parse a multipart form body with a field called `Pass`, you would use a struct field with `form:"pass"`.
 
 ```go title="Signature"
 func (b *Bind) MultipartForm(out any) error
 ```
 
 ```go title="Example"
-// Field names should start with an uppercase letter
```
Comment (ReneWerner87) [inline docs/api/bind.md:166]: i like this comment, don´t remove it

File: docs/api/bind.md [hunk 7]
```diff
@@ -181,25 +180,25 @@ app.Post("/", func(c fiber.Ctx) error {
...
 
-It is important to specify the correct struct tag based on the content type to be parsed. For example, if you want to parse an XML body with a field called Pass, you would use a struct field of `xml:"pass"`.
+It is important to specify the correct struct tag based on the content type to be parsed. For example, if you want to parse an XML body with a field called `Pass`, you would use a struct field with `xml:"pass"`.
 
 ```go title="Signature"
 func (b *Bind) XML(out any) error
 ```
 
 ```go title="Example"
-// Field names should start with an uppercase letter
```
Comment (ReneWerner87) [inline docs/api/bind.md:202]: we should keep this comment

File: docs/api/bind.md [hunk 8]
```diff
@@ -402,64 +416,66 @@ func (b *Bind) Custom(name string, dest any) error
...
+func (cb *customBinder) Parse(c fiber.Ctx, out any) error {
+    // parse YAML body
     return yaml.Unmarshal(c.Body(), out)
 }
+
 // Register custom binder
-app.RegisterCustomBinder(customBinder)
+app.RegisterCustomBinder(&customBinder{})
 
-// curl -X POST http://localhost:3000/custom -H "Content-Type: application/yaml" -d "name: John"
```
Comment (ReneWerner87) [inline docs/api/bind.md:422]: this inline example is missing

File: docs/api/bind.md [hunk 9]
```diff
@@ -578,19 +601,19 @@ app.Post("/", func(c fiber.Ctx) error {
...
+- `string`
+- A slice of the above types. Use `|` to separate slice items.
+- A pointer to one of the above types (**pointers to slices and slices of pointers are not supported**).
 
 ```go title="Example"
 type Person struct {
-    Name     string     `query:"name,default:john"`
-    Pass     string     `query:"pass"`
-    Products []string   `query:"products,default:shoe|hat"`
+    Name     string   `query:"name,default=John"`
```
Comment (ReneWerner87) [inline docs/api/bind.md:614]: i think it is using colon for the default value

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
@gaby can you check the hints

**Comment by gaby** [comment]
> @gaby can you check the hints

👍 Done

File: docs/api/redirect.md [hunk 10]
```diff
@@ -6,14 +6,13 @@ sidebar_position: 5
 toc_max_heading_level: 5
 ---
 
-Is used to redirect the ctx(request) to a different URL/Route.
+The redirect package is used to redirect the context (request) to a different URL or route.
```
Comment (efectn) [inline docs/api/redirect.md:9]: I think package is not convenient here. Maybe we can replace it to the Redirect methods?

File: docs/api/bind.md [hunk 11]
```diff
@@ -600,13 +626,16 @@ app.Get("/", func(c fiber.Ctx) error {
         return err
     }
 
-    log.Println(p.Name)        // john
-    log.Println(p.Pass)        // doe
-    log.Println(p.Products)    // ["shoe,hat"]    
+    log.Println(p.Name)     // John
```
Comment (ReneWerner87) [inline docs/api/bind.md:629]:
```suggestion
    log.Println(p.Name)     // john
```

File: docs/api/redirect.md (hunk 10 above)
Comment (ReneWerner87) [inline docs/api/redirect.md:9]:
```suggestion
The redirect methods are used to redirect the context (request) to a different URL or route.
```

Comment (gaby) [reply docs/api/redirect.md:9]: Yeah, this is a bug from ChatGPT 😂 it shouldn't say package

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**